src/main/java/com/github/manolo8/darkbot/core/objects/facades/DiminishQuestMediator.java

```diff
@@ -0,0 +1,59 @@
+package com.github.manolo8.darkbot.core.objects.facades;
+
+import com.github.manolo8.darkbot.core.itf.Updatable;
+import com.github.manolo8.darkbot.core.objects.gui.DiminishQuestGui;
+import eu.darkbot.api.API;
+import eu.darkbot.api.managers.QuestAPI;
+
+import static com.github.manolo8.darkbot.Main.API;
+
+public class DiminishQuestMediator extends Updatable implements API.Singleton {
+    private State state = State.NOT_EXIST;
+    private final DiminishQuestGui diminishQuestGui;
+    private final QuestProxy.Quest diminishQuest = new QuestProxy.Quest();
+
+    public DiminishQuestMediator(DiminishQuestGui diminishQuestGui) {
+        this.diminishQuestGui = diminishQuestGui;
+    }
+
+    @Override
+    public void update() {
+        if (address == 0) {
+            state = State.NOT_EXIST;
+        } else if (!"diminish_quests".equals(API.readString(address, 0x38))) {
```

Comment:
given the string caching that we do, and how otherwise string reading is an expensive operation, i'd say you can probably avoid doing this check at all.

Look into implementing separate update(addr) and update() methods, where the former invalidates some stuff and checks for this string, and this method checks if the address of the string it points to changes, if that is even required.

Is this an actual requirement or just a memory-bug fixing strategy?

Reply:
It's solution that allow be sure that flash object was not removed by GC.
It's issue of flashmap updating where key removing from collection, bot foget about this instance and it caused that the existing memory pointer (address) is not actual can be reused by another object.

Reply:
that's something that the bot's dictionary for guis should already be handling, if the object stops being in the map it should be reset to address 0
i believe it was a bug that the older implementations for dictionary weren't doing it, but the new flashmap should be doing that just fine

Reply:
It's great. So I'll remove this condition.

Reply:
Done